sympy/polys/matrices/sdm.py

```diff
@@ -66,6 +66,13 @@ def to_ddm(M):
     def zeros(cls, shape, domain):
         return cls({}, shape, domain)
 
+    @classmethod
+    def ones(cls, shape, domain):
+        one = domain.one
+        m, n = shape
+        sdm = {i: {j: one for j in range(n)} for i in range(m)}
+        return cls(sdm, shape, domain)
```

Comment:
You should run some timings to check this but I would imagine that for a large shape it is quicker to create a dict once and then copy it many times rather than use a dict comprehension for over j for each i:
```
row = dict(zip(range(n), [one]*n))
sdm = {i: row.copy() for i in range(m)}
```

Reply:
I ran some benchmarks

```python
from sympy.polys.matrices.sdm import SDM
def ones_new(shape, domain):
    m, n = shape
    one = domain.one
    row = dict(zip(range(n), [one]*n))
    sdm = {i: row.copy() for i in range(m)}
def ones_current(shape, domain):
    one = domain.one
    m, n = shape
    sdm = {i: {j: one for j in range(n)} for i in range(m)}
```
```python
from sympy import QQ
for i in range(100,1000, 100):
    %time ok = ones_new((i, i), QQ)


CPU times: user 0 ns, sys: 987 µs, total: 987 µs
Wall time: 1 ms
CPU times: user 601 µs, sys: 1.65 ms, total: 2.25 ms
Wall time: 2.27 ms
CPU times: user 0 ns, sys: 2.56 ms, total: 2.56 ms
Wall time: 2.57 ms
CPU times: user 3.04 ms, sys: 4 ms, total: 7.04 ms
Wall time: 6.97 ms
CPU times: user 0 ns, sys: 6.82 ms, total: 6.82 ms
Wall time: 6.78 ms
CPU times: user 2.64 ms, sys: 4.69 ms, total: 7.33 ms
Wall time: 7.27 ms
CPU times: user 0 ns, sys: 14.4 ms, total: 14.4 ms
Wall time: 14 ms
CPU times: user 11 ms, sys: 9.31 ms, total: 20.3 ms
Wall time: 19.8 ms
CPU times: user 2.5 ms, sys: 16.2 ms, total: 18.7 ms
Wall time: 18.6 ms

```

```python
for n in range(100, 1000, 100):
    %time ok = ones_current((n, n), QQ)


CPU times: user 1.48 ms, sys: 178 µs, total: 1.66 ms
Wall time: 1.67 ms
CPU times: user 5.5 ms, sys: 0 ns, total: 5.5 ms
Wall time: 5.31 ms
CPU times: user 7.18 ms, sys: 0 ns, total: 7.18 ms
Wall time: 7.12 ms
CPU times: user 13.2 ms, sys: 49 µs, total: 13.2 ms
Wall time: 13.2 ms
CPU times: user 20.1 ms, sys: 0 ns, total: 20.1 ms
Wall time: 19.9 ms
CPU times: user 26.5 ms, sys: 0 ns, total: 26.5 ms
Wall time: 26.6 ms
CPU times: user 21.3 ms, sys: 15.8 ms, total: 37.1 ms
Wall time: 37.1 ms
CPU times: user 31.7 ms, sys: 12.1 ms, total: 43.8 ms
Wall time: 43.8 ms
CPU times: user 42.5 ms, sys: 12.5 ms, total: 55 ms
Wall time: 55 ms

```

Your method is a little bit faster